### DISCUSSION TITLE
A transation across multiple entities

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2620] [Q&A]
I need to insert some data every second, actually 10 insertions/updates per machine per second (the number of machines can be anywhere between 10-100). These insertions/updates are all for different entities.

Based on the [docs](https://mikro-orm.io/docs/transactions/#approach-2-explicitly), I was thinking if I could add a cronjob where I would:
1. `em.commit()` all insertions/updates done from the previous cronjob run (if any);
2. `em.begin()` a new transaction.

I see one problem with this only: if any one insertion/update fails, all of them will be reverted (`em.rollback()`).

Is this approach possible at all? Are there any other problems you see? Is there any better approach to this to improve the performance.

**Comment by B4nan** [comment]
Depends on your use case. It sounds more and more like you would be better of without using identity map or implement some kind of queuing system yourself on top of it. Don't expect the ORM to do this for you, proper way is to have separate contexts for each handler (which from what I understood you don't want).

A transaction context is always the EM and its UoW/identity map. It will always be about what the EM is managing. It you want to control this, you can use EM forks.
